cmd/nitrite/main_test.go

```diff
@@ -57,32 +59,64 @@ func runNitriteWithEnv(
 }
 
 func TestExec(t *testing.T) {
-	t.Run("nitro attestation", func(t *testing.T) {
-		// given
-		cmd := "go run main.go"
-		cmd += " -attestation " + internal.NitroAttestationB64
-
-		// when
-		outDoc, err := runNitriteWithEnv(cmd, nil)
-
-		// then
-		require.NoError(t, err)
-		assert.NotEmpty(t, outDoc)
-	})
-
-	t.Run("debug nitro attestation", func(t *testing.T) {
-		// given
-		cmd := "go run main.go"
-		cmd += " -attestation " + internal.DebugNitroAttestationB64
-		cmd += " -allowdebug"
+	happyPathTests := []struct {
+		name        string
+		attestation string
+		debug       bool
+		selfSigned  bool
+		time        time.Time
+		wantPCR0    string
+	}{
+		{
+			name:        "happy path - nitro",
+			attestation: internal.NitroAttestationB64,
+			debug:       false,
+			selfSigned:  false,
+			time:        internal.NitroAttestationTime,
+			wantPCR0:    "5ypGyoCiYPsEShJUQvDH4zGBO8uvlyTZ84V3WJknZvLWVxCieqlK45Sd1U58n+hq",
+		},
+		{
+			name:        "happy path - debug nitro",
+			attestation: internal.DebugNitroAttestationB64,
+			debug:       true,
+			selfSigned:  false,
+			time:        internal.DebugNitroAttestationTime,
+			wantPCR0:    "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
+		},
+		{
+			name:        "happy path - self signed",
+			attestation: internal.SelfSignedAttestationB64,
+			debug:       false,
+			selfSigned:  true,
+			time:        internal.SelfSignedAttestationTime,
+			wantPCR0:    "MDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAwMDAw",
+		},
+	}
+	for _, tt := range happyPathTests {
+		t.Run(tt.name, func(t *testing.T) {
+			// given
+			cmd := "go run main.go"
+			cmd += " -attestation " + tt.attestation
+			if tt.debug {
+				cmd += " -allowdebug"
+			}
+			if tt.selfSigned {
+				cmd += " -allowselfsigned"
+			}
 
 		// when
 		outDoc, err := runNitriteWithEnv(cmd, nil)
 
-		// then
-		require.NoError(t, err)
-		assert.NotEmpty(t, outDoc)
-	})
+			// then
+			require.NoError(t, err)
+			assert.Equal(t, tt.time, outDoc.CreatedAt().UTC())
+			assert.Equal(t, tt.wantPCR0, base64.StdEncoding.EncodeToString(outDoc.PCRs[0]))
+			gotDebug, err := outDoc.Debug()
+			require.NoError(t, err)
+			assert.Equal(t, tt.debug, gotDebug)
+			assert.NotEmpty(t, outDoc.PublicKey)
```

Comment:
The [nitro docs](https://docs.aws.amazon.com/enclaves/latest/user/verify-root.html#doc-def) state that the public key is an optional parameter. Given that it *could* be empty, should we actually be making this assertion?

Reply:
That's a good point. Dave suggested that we test the structure and content of the doc coming back and we do put a public key in all the attestations. Having said that, you're right, that its not required and so this test might not age well as a part of the library, where the testdata might change later on. I'll take this check out.